### PR TITLE
added realizetion of Build patter

### PR DESCRIPTION
**Pull request by MarberyUA**

### CODE REVIEW

File: src/main/java/core/basesyntax/Plane.java [hunk 1]
```diff
@@ -8,4 +8,74 @@
...
 public class Plane {
 
+    private Integer power;
+    private Integer size;
+    private String color;
+    private Integer seats;
+    private String model;
+
+    public Plane(PlaneBuilder builderForPlane) {
+        this.model = builderForPlane.model;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Plane.java:19]: you should use all fields

File: src/main/java/core/basesyntax/Plane.java [hunk 2]
```diff
@@ -8,4 +8,74 @@
...
+        private Integer size;
+        private String color;
+        private Integer seats;
+        private String model;
+
+        private PlaneBuilder() {
+        }
+
+        public PlaneBuilder setPower(Integer power) {
+            Plane.this.power = power;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Plane.java]: which fields should you set?

File: src/main/java/core/basesyntax/Plane.java [hunk 3]
```diff
@@ -8,4 +8,74 @@
...
+    }
+
+    public class PlaneBuilder {
+        private Integer power;
+        private Integer size;
+        private String color;
+        private Integer seats;
+        private String model;
+
+        private PlaneBuilder() {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Plane.java]: why did you make it private?

File: src/main/java/core/basesyntax/Plane.java [hunk 4]
```diff
@@ -8,4 +8,74 @@
...
+            return this;
+        }
+
+        public PlaneBuilder setModel(String model) {
+            Plane.this.model = model;
+            return this;
+        }
+
+        public Plane build() {
+            return Plane.this;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Plane.java]: I can't see the `building` process :)

File: src/main/java/core/basesyntax/Plane.java [hunk 5]
```diff
@@ -8,4 +8,80 @@
 
 public class Plane {
 
+    private Integer power;
+    private Integer size;
+    private String color;
+    private Integer seats;
+    private String model;
+
+    public Plane(PlaneBuilder builderForPlane) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Plane.java]: we should have access for this constructor only within this class

Comment (MarberyUA) [reply src/main/java/core/basesyntax/Plane.java]: Yup, sorry i had watched wrong realizetion of Builder pathern

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**